tripal/src/TripalPubLibrary/TripalPubLibraryBase.php

```diff
@@ -97,4 +97,30 @@ public function __construct(array $configuration, $plugin_id, $plugin_definition
     $this->fileretriever = $fileretriever;
   }
 
+  /**
+   * Validates whether XML is valid or not.
+   *
+   * @param string $xml
+   *   The XML to be checked.
+   *
+   * @return bool
+   *   Return TRUE if valid, FALSE if not valid.
+   */
+  protected function xmlIsValid(string $xml): bool {
+    $valid = TRUE;
+
+    libxml_use_internal_errors(TRUE);
+    $doc = simplexml_load_string($xml);
+    if ($doc === FALSE) {
+      $valid = FALSE;
+    }
+    else {
```

Comment:
```suggestion
    // Attempt to load the XML.
    $doc = simplexml_load_string($xml);
    // If SimpleXML fails to parse the XML string then it will return FALSE.
    if ($doc === FALSE) {
      $valid = FALSE;
    }
    // If not, we will check for any errors logged during parsing.
    else {
```